tutorials/advanced-development/advanced-pieces-topics/adding-filters.md

```diff
@@ -1,40 +1,212 @@
+
+However, keep in mind that if you change the slug someone's bookmarked links might break. So it's up to you whether to use `_jobs` (for the `_id`) or `jobs` (for the `slug`).
+{% endhint %}
+
+Now you can filter people by job:
+
+{% code-tabs %}
+{% code-tabs-item title="lib/modules/people-pages/index.html" %}
+```markup
+{# Link to all the tags, adding a parameter to the query string #}
+<ul class="job-filters">
+  {% for job in data.piecesFilters.jobs %}
+    <li><a href="{{ data.url | build({ jobs: job.value }) }}">{{ job.label }}</a></li>
+  {% endfor %}
+</ul>
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+{% hint style="info" %}
+Notice that this template looks exactly like the one for tags. That's intentional. You could use a single Nunjucks macro for both.
+{% endhint %}
+
+### Filtering on multiple values
+
+You're not restricted to filtering on a single value for a join, or for a `tags` field. If the query string looks like this:
+
+```
+?jobs[]=anointer&jobs[]=flosser
+```
+
+You'll see people with *either* job.
+
+{% hint style="info" %}
+If you want to be more restrictive and only display results that have *all* of the specified values, add `And` to the filter name&mdash;in both the `piecesFilters` array (in the module's `index.js`) and the template references. As before, you can do this with `_` for `_id`, or without for `slug`.For instance, `_jobsAnd` expects ids, while `jobsAnd` expects slugs.
+
+However, **keep in mind this usually is very frustrating for users because they will rarely get any matches.** We recommend the default "or" behavior.
+{% endhint %}
+
+Here's how to build query strings that contain arrays in your template:
+
+{% code-tabs %}
+{% code-tabs-item title="lib/modules/people-pages/index.html" %}
+```markup
+<ul class="job-filters">
+  {% for job in data.piecesFilters.jobs %}
+    {% if apos.utils.contains(data.query.jobs, job.value) %}
+      <a href="{{ data.url | build({ jobs: { $pull: job.value } }) }}" class="current">
+    {% else %}
+      <a href="{{ data.url | build({ jobs: { $addToSet: job.value } }) }}">
+    {% endif %}
+      {{ job.label }}
+    </a>
+  {% endfor %}
+</ul>
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+{% hint style="info" %}
+**"What's going on in this code?"** Like before, we are using the `build` filter to add and remove query parameters. However, this time, we are using the special `$pull` operator to remove a job from the array without removing the others, and using the special `$addToSet` operator to add a job to the array. In this way, we can manage filter URLs like `/people?jobs[]=doctor&jobs[]=technician` with very little effort.
+{% endhint %}
+
+## Next Steps
+
+If you want to try out some other uses of filters, you can see how to [integrate them with Apostrophe's site management interface](adding-columns.md), or you can continue on to learn how to use AJAX to add dynamic behavior to your site.
```

Comment:
Change that to "use AJAX to avoid full page refreshes when filters are used."

Reply:
Updated.